beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -302,7 +302,15 @@ func (s *Service) deleteBlockFromPendingQueue(slot uint64, b *ethpb.SignedBeacon
 
 	newBlks := make([]*ethpb.SignedBeaconBlock, 0, len(blks))
 	for _, blk := range blks {
-		if ssz.DeepEqual(blk, b) {
+		expectedRoot, err := blk.HashTreeRoot()
+		if err != nil {
+			log.Fatal(err)
+		}
+		receivedRoot, err := b.HashTreeRoot()
+		if err != nil {
+			log.Fatal(err)
+		}
+		if bytes.Equal(expectedRoot[:], receivedRoot[:]) {
```

Comment:
This will be simply too expensive / wasteful to do these HTR at runtime. We'll need to copy over DeepEqual to our code and continue to use it in these comparisons. 

The comparison would be treated the same for `nil == empty slice`, see:  https://github.com/prysmaticlabs/prysm/issues/7845#issuecomment-758610675

Reply:
@0xKiwi I was working on copying the `ssz.DeepEqual` and realize you have a full PR. See if it is useful:

https://github.com/prysmaticlabs/prysm/pull/8258

Reply:
I'm withholding my approval until #8258 is merged and this issue resolved.